unyt/dask_array.py

```diff
@@ -0,0 +1,620 @@
+            return _create_with_quantity(dask_result, unyt_result)
+        return dask_result
+
+    return wrapper
+
+
+def _special_dec(the_func):
+    # decorator for special operations like __mul__ , __truediv__, which will
+    # bypass __getattribute__
+    def wrapper(self, *args, **kwargs):
+        funcname = the_func.__name__
+        ufunc = getattr(ua.unyt_quantity, funcname)
+        ufunc_args = (self,) + args
+        args, unyt_result = _prep_ufunc(ufunc, *ufunc_args, extract_dask=True, **kwargs)
+
+        # remove the first arg, cause we need to supply self as first arg
+        _ = args.pop(0)
+        daskresult = the_func(self, *args, **kwargs)
+
+        if hasattr(unyt_result, "units"):
+            return _create_with_quantity(daskresult, unyt_result)
+        return daskresult  # __lt__, __le__, etc. will hit this
+
+    return wrapper
+
+
+class unyt_dask_array(_dask_Array):
+    """
+    a dask.array.core.Array subclass that tracks units. Easiest to use the
+    unyt_from_dask helper function to generate new instances.
+
+    Parameters
+    ----------
+
+    All parameters are those for dask.array.core.Array
+
+    """
+
+    def __new__(
+        clss,
+        dask_graph,
+        name,
+        chunks,
+        dtype=None,
+        meta=None,
+        shape=None,
+        units=None,
+        registry=None,
+        bypass_validation=False,
+        input_units=None,
+        unyt_name=None,
+    ):
+
+        # get the base dask array
+        obj = super(unyt_dask_array, clss).__new__(
+            clss,
+            dask_graph,
+            name,
+            chunks,
+            dtype,
+            meta,
+            shape,
+        )
+
+        # attach our unyt sidecar quantity
+        dtype = obj.dtype
+        obj._unyt_array = ua.unyt_array(
+            [1.0],
+            units,
+            registry,
+            dtype,
+            bypass_validation,
+            input_units,
+            unyt_name,
+        )
+
+        obj.units = obj._unyt_array.units
+        obj.unyt_name = obj._unyt_array.name
+
+        # set the unit conversion attributes so they are discoverable. no name
+        # conflicts for now, but this could be an issue if _unyt_funcs_to_track
+        # is expanded.
+        for attr in _unyt_funcs_to_track:
+            setattr(obj, attr, getattr(obj._unyt_array, attr))
+
+        return obj
+
+    def _elemwise(self, ufunc, *args, **kwargs):
+        args, unyt_result = _prep_ufunc(ufunc, *args, **kwargs)
+        return _post_ufunc(super()._elemwise, unyt_result)(ufunc, *args, **kwargs)
+
+    def __array_ufunc__(self, numpy_ufunc, method, *inputs, **kwargs):
+        inputs, unyt_result = _prep_ufunc(
+            numpy_ufunc, *inputs, extract_dask=True, **kwargs
+        )
+        wrapped_func = _post_ufunc(super().__array_ufunc__, unyt_result)
+        return wrapped_func(numpy_ufunc, method, *inputs, **kwargs)
+
+    def __array_function__(self, func, types, args, kwargs):
```

Comment:
We should really make sure there are no flagrant incompatibilities with #200. 
I think #200 is more fundamental so I would argue it should go in first if possible, but I know it's a big one too. At a minimum, we should try to run all tests in a temporary merger branch.

Reply:
Update: I checked and found no real merge conflict and 0 test failures